x-pack/plugins/observability_solution/logs_explorer/public/utils/convert_discover_app_state.ts

```diff
@@ -58,18 +75,34 @@ export const getDiscoverAppStateFromContext = (
   filters: cloneDeep(displayOptions.filters),
 });
 
+export const getDiscoverColumnsWithFallbackFieldsFromDisplayOptions = (
+  displayOptions: DisplayOptions
+): DiscoverAppState['columns'] =>
+  displayOptions.grid.columns.flatMap((column) => {
+    if (column.type === 'document-field') {
+      return column.field;
+    } else {
+      return SMART_FALLBACK_FIELDS[column.name];
+    }
```

Comment:
Else is not needed here

```suggestion
    
      return SMART_FALLBACK_FIELDS[column.name];

```